### PR TITLE
Argon2id type support

### PR DESCRIPTION
**Pull request by markfejes**
Hey there!

As argon2's latest changelog says they implemented the argon2id type. This pull request adds support for your already wonderful module both verify and hashing of course. This means, that the argon2 includes have to be updated to support argon2id.

Please review it @ranisalt, if I made any mistakes!

Thank you very much!

### CODE REVIEW

File: src/argon2_node.cpp [hunk 1]
```diff
@@ -126,7 +127,7 @@ VerifyWorker::VerifyWorker(std::string&& hash, std::string&& plain):
 
 void VerifyWorker::Execute()
 {
-    auto type = (hash.at(7) == 'd') ? Argon2_d : Argon2_i;
+    auto type = (hash.at(7) == 'd') ? Argon2_d : ( hash.at(8) == 'd' ? Argon2_id : Argon2_i );
```
Comment (ranisalt) [inline src/argon2_node.cpp]: Better to switch to if/elseif/else structure, ternary operator is very ugly here.

Comment (markfejes) [inline src/argon2_node.cpp]: True that :) I thought you like to keep the code really small

Comment (ranisalt) [inline src/argon2_node.cpp]: Well, readability matters hahahah

**Comment by markfejes** [comment]
@ranisalt Can we merge this today? :) If so, then could you bump the version afterwards?

File: index.js [hunk 2]
```diff
@@ -56,7 +63,7 @@ module.exports = {
   },
 
   verify(hash, plain) {
-    if (!/^\$argon2[di](\$v=\d+)?\$m=\d+,t=\d+,p=\d+(?:\$[\w+/]+){2}$/.test(hash)) {
+    if (!/^\$argon2[di]{1,2}(\$v=\d+)?\$m=\d+,t=\d+,p=\d+(?:\$[\w+/]+){2}$/.test(hash)) {
```
Comment (ranisalt) [inline index.js]: It's better to list all possibilities. `(i|d|id)`

**Pull request merged**